### PR TITLE
middle-router should ignore link clicks that originated from anchor t…

### PR DESCRIPTION
**Pull request by shichongrui**
…ags that have a target or download attribute. These links have intended behavior outside of the context of middle-router

### CODE REVIEW

File: lib/history.js [hunk 1]
```diff
@@ -143,7 +143,8 @@ function didClick (router, event) {
   while (target && target.nodeName.toLowerCase() !== 'a' && target !== html) {
     target = target.parentNode
   }
-  if (!target || target.nodeName.toLowerCase() !== 'a') return
+  if (!target || target.nodeName.toLowerCase() !== 'a' ||
+      target.hasAttribute('target') || target.hasAttribute('download')) return
```
Comment (thetalecrafter) [inline lib/history.js]: can you add the `rel` attribute as well? `rel=external` is a valid way to say the link leaves site.

Comment (shichongrui) [inline lib/history.js]: Should it be specific to rel=external or just rel at all?

File: test/links.js [hunk 2]
```diff
@@ -34,6 +37,63 @@ describe('Router#routeLinks', () => {
...
+        assert.fail('should not route due to target attribute')
+      })
+
+    history.replaceState(null, document.title, '/start')
+    await router.start()
+
+    clickTo('#/linked/location', false, { target: '_self' })
+    await router.routing
+
+    router.stop()
```
Comment (thetalecrafter) [inline test/links.js]: Can you assert the `location.hash` was updated, so we know that the default behavior was not prevented?

Comment (thetalecrafter) [inline test/links.js]: Actually nevermind. A synthetic click like this won't trigger a url change because it didn't come from a user.

**Pull request merged**